feat(terra/policy): add policy for tags, versioning and release field (#137)
FyraLabs/devdocs: pages/terra/policy.mdx

    8. `desktops/` (applications specific to a Desktop Environment)
    9. `apps/` (GUI applications)
    10. `langs/` (software written in/for a specific programming language)
- When there are ambiguous or unspecified parts in our policies/guidelines, you SHOULD refer to [Fedora's packaging guidelines](https://docs.fedoraproject.org/en-US/packaging-guidelines).
- Low-quality contributions will be declined at the team's discretion.
- Do not include work-in-progress code or untested patches. See https://dont-ship.it/.

## Maintenance Policy

- Anyone MAY make a PR (pull request) to an already existing package.
- Unmaintained packages may be removed at any time. Before doing so, we will create a GitHub issue to gauge community interest.
- Terra will only support the latest version of Fedora, if a package doesn't build on the next version of Fedora, we might hold it back.

## Naming and Versioning Scheme

This section describes the guidelines for naming and versioning packages correctly in both Terra and Terra Extras.

### Tags

> First of all, the term "_tag_" is used to refer to different versions of the same software, as in their release schedule / model.

Reviewer: ```suggestion
First of all, the term "_tag_" is used to refer to different versions of the same software, as in their release schedule/model.
```

  |VCS| ID |
  |-----|-------|
  |Git| `git`|
  |SVN| `svn`|
  |CVS| `cvs`|
  |Bazaar| `bzr`|
  |Mercurial| `hg`|
  |Perforce (Helix)| `p4`|
  |Pijul| `pjl` |
  |Azure DevOps| `tfs` |
  |darcs| `darc`|
  |Fossil|`fsl`|
</div>

Packages SHOULD follow one of the version formats below:

```rpmspec
## (a)
Version:  %{latest_stable_version}
## (b)
Version:  %{latest_stable_version}^%{commit_date}git%{shortcommit}
## (c)
Version:  %{unreleased_version}~%{commit_date}git%{shortcommit}
```

`git` SHOULD be replaced with the correct _VCS identifier_ for **(b)** and **(c)** as listed on the right.

**(b)** and **(c)** MUST be used with `labels { nightly = 1 }{:hcl}` in `anda.hcl` unless there is no `update.rhai`,
as they rely on the assumption that there is 1 `%shortcommit` for each `%commit_date` (and vice versa).

`%unreleased_version` refers to the expected version number for the next stable (or a more stable _tag_ than `nightly`) release,
while `%latest_stable_version` refers to the current latest released version number, preferably stable.

**(c)** SHOULD be preferred over **(b)** if possible, since having a `%unreleased_version` can indicate to users

> breaking changes / new features depending on the versioning system (usually SemVer).

Reviewer: ```suggestion
breaking changes/new features depending on the versioning system (usually SemVer).
```

Without approval from the team, packages MUST use the format `Release: 1%?dist{:rpmspec}` (or `1%{?dist}{:rpmspec}`).

The release number SHOULD be incremented if changes are made to the package, unless the `Version:{:rpmspec}` changes.

### Development and Shared Libraries

Shared library packages SHOULD be suffixed with `-libs`.
These packages SHOULD NOT contain anything **except**:

```rpmspec
%_libdir/*.so.*
%doc …
%license …
```

Development library packages SHOULD be suffixed with `-devel`.
These packages SHOULD NOT contain anything **except**:

```rpmspec
%doc …
%license …
%_includedir/*
# ^ source files, depending on the language, could be in other locations
%_libdir/*.so
%_libdir/pkgconfig/%{name}.pc
# and other development files (.vapi, .typelib, .gir, …)
```

### Naming

Prefix/Suffix the packages in the following order
(that means nightly fonts should end with `-fonts%.nightly`)

- Fonts MUST be suffixed with `-fonts`.
- Shared libraries SHOULD be either suffixed with `-libs` or prefixed with `lib`.
- Development libraries SHOULD be suffixed with `-devel`.
- If necessary, suffix `-bash-completion` (or other shells) and `-doc`.

> - Postfix a _tag_ if necessary (e.g. `%.nightly`).

Reviewer 2: We should really come up with a better notation for tags, because I fear using `%` may cause some footguns down the line due to the fact that it's an RPM macro notator

Author: then it's by design a bug that rpm should fix 🚎

## Security Policy

Our security policy is documented in the [SECURITY.md file](https://github.com/terrapkg/packages/blob/frawhide/SECURITY.md).

## Code of Conduct

Our code of conduct is documented in the [CODE_OF_CONDUCT.md file](https://github.com/terrapkg/packages/blob/frawhide/CODE_OF_CONDUCT.md).

## Lifecycle Policy

Our support lifecycle is documented in the [lifecycle page](/terra/lifecycle).

## Packaging Guidelines

Terra uses a slightly different packaging guideline for packages from Fedora's. This article is to outline the specific different guidelines in Terra.

### Interpreted Languages (Python, Ruby, etc.)

All packages using interpreted languages follow the traditional Fedora guidelines, with all dependencies **_must_** be packaged individually as they are considered runtime dependencies.

### Dynamically compiled languages (C, C++, Vala)

Dependencies must also be packaged separately if they are required for runtime.

### Statically compiled languages

> As Terra does not strictly follow Fedora's reproducibility requirements, and we do not want large amounts of development library languages for them, packagers **should** vendor dependencies on build time.

Reviewer 2: i just noticed this typo 3 years later

```suggestion
As Terra does not strictly follow Fedora's reproducibility requirements, and we do not want large amounts of development libraries for them, packagers **should** vendor dependencies on build time.
```

Terra's mock sandbox has networking enabled, so builders can download the dependencies directly on build time.

#### Usages of the `mold` linker

> We encourage the use of `mold`, which may speed up build times especially in large projects. You may enable it in C/++ projects by adding `-fuse-ld=mold` in `CFLAGS`/`CXXFLAGS`.

Reviewer 2: We probably wanna link it

```suggestion
We encourage the use of [`mold`](https://github.com/rui314/mold), which may speed up build times especially in large projects. You may enable it in C/++ projects by adding `-fuse-ld=mold` in `CFLAGS`/`CXXFLAGS`.
```

%cargo_install
```

#### Go

Use the `go` compiler directly instead of using Fedora's Go macros, as this will generate offline dependencies that we do not want to package.

> Alternative, use the `_online` macro variants:

Reviewer 2: Write an example for using the macro variants, in case someone is dense enough to not figure out what it's actually called

Reviewer 2: typo
```suggestion
Alternatively, use the `_online` macro variants:
```
